libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/RouteRefreshCallback.java

```diff
@@ -0,0 +1,43 @@
+package com.mapbox.services.android.navigation.v5.navigation;
+
+import com.mapbox.api.directions.v5.models.DirectionsRoute;
+import com.mapbox.api.directionsrefresh.v1.models.DirectionsRefreshResponse;
+
+import retrofit2.Call;
+import retrofit2.Callback;
+import retrofit2.Response;
+
+class RouteRefreshCallback implements Callback<DirectionsRefreshResponse> {
+  private final RouteAnnotationUpdater routeAnnotationUpdater;
+  private final DirectionsRoute directionsRoute;
+  private final int legIndex;
+  private final RefreshCallback refreshCallback;
+
+  RouteRefreshCallback(DirectionsRoute directionsRoute,
+                       int legIndex, RefreshCallback refreshCallback) {
+    this(new RouteAnnotationUpdater(), directionsRoute, legIndex, refreshCallback);
+  }
+
+  RouteRefreshCallback(RouteAnnotationUpdater routeAnnotationUpdater, DirectionsRoute directionsRoute,
+                       int legIndex, RefreshCallback refreshCallback) {
+    this.routeAnnotationUpdater = routeAnnotationUpdater;
+    this.directionsRoute = directionsRoute;
+    this.legIndex = legIndex;
+    this.refreshCallback = refreshCallback;
+  }
+
+  @Override
+  public void onResponse(Call<DirectionsRefreshResponse> call, Response<DirectionsRefreshResponse> response) {
+    if (response.body() == null || response.body().route() == null) {
```

Comment:
Could `route` be ever `null`? 🤔 For Directions, `routes` is marked as `@NonNull`.